opamp/observiq/observiq_client.go

```diff
@@ -0,0 +1,195 @@
+// Copyright  observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package observiq contains OpAmp structures compatible with the observiq client
+package observiq
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+
+	"github.com/observiq/observiq-otel-collector/collector"
+	"github.com/observiq/observiq-otel-collector/opamp"
+	"github.com/open-telemetry/opamp-go/client"
+	"github.com/open-telemetry/opamp-go/client/types"
+	"github.com/open-telemetry/opamp-go/protobufs"
+	"go.uber.org/zap"
+)
+
+var (
+	// ErrUnsupportedURL is error returned when creating a client with an unsupported URL scheme
+	ErrUnsupportedURL = errors.New("unsupported URL")
+)
+
+// Ensure interface is satisfied
+var _ opamp.Client = (*Client)(nil)
+
+// Client represents a client that is connected to Iris via OpAmp
+type Client struct {
+	opampClient   client.OpAMPClient
+	logger        *zap.SugaredLogger
```

Comment:
Why is this logger sugar'ed? It looks like we're always using the non-sugared, field based methods anyways.

Reply:
I think this is a hold over from the agent code I copied. I'll factor it out.

Reply:
Ohh looks like the opamp library logging interface isn't fulfilled by a non-sugared logger. I can just sugar it when it's passed into the implementation